spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/VectorSearch.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.mongodb.repository;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.data.mongodb.core.aggregation.VectorSearchOperation;
+
+/**
+ * Annotation to declare Vector Search queries directly on repository methods. Vector Search queries are used to search
+ * for similar documents based on vector embeddings typically returning
+ * {@link org.springframework.data.domain.SearchResults} and limited by either a
+ * {@link org.springframework.data.domain.Score} (within) or a {@link org.springframework.data.domain.Range} of scores
+ * (between).
+ * <p>
+ * Vector search must define an index name using the {@link #indexName()} attribute. The index must be created in the
+ * MongoDB Atlas cluster before executing the query. Any misspelling of the index name will result in returning no
+ * results.
+ * <p>
+ * When using pre-filters, you can either define {@link #filter()} or use query derivation to define the pre-filter.
+ * {@link org.springframework.data.domain.Vector} and distance parameters are considered once these are present. Vector
+ * search supports sorting and will consider {@link org.springframework.data.domain.Sort} parameters.
+ *
+ * @author Mark Paluch
+ * @since 5.0
+ * @see org.springframework.data.domain.Score
+ * @see org.springframework.data.domain.Vector
+ * @see org.springframework.data.domain.SearchResults
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.ANNOTATION_TYPE })
+@Documented
+@Query
+@Hint
+public @interface VectorSearch {
+
+	/**
+	 * Configuration whether to use ANN or ENN for the search. ANN is the default.
+	 *
+	 * @return the search type to use.
+	 */
+	VectorSearchOperation.SearchType searchType() default VectorSearchOperation.SearchType.ENN;
```

Comment:
Documentation (default: ANN) contradicted by implementation (default: ENN).
MongoDB itself will use `ANN` by default (so the doc) is closer to what happens than the actual default value.

I think the MongoDB restrictions here do not allow much flexibility regarding defaulting of values. Like on the one hand `ANN` ist stated as default but requires `numCandidates` while the `exact` flag is labelled optional.  

We could argue `DEFAULT` is a viable option here. Checking the `numCandidates` attribute, using `ENN` if no value present, `ANN` otherwise.